### PR TITLE
Add bottlejs@1.4.0 w/ npm auto-update config

### LINKED ISSUE #9149: [Request] Add bottlejs
**Library name:** bottlejs
**Git repository url:** https://github.com/young-steveo/bottlejs
**npm package url(optional):** bottlejs

### PR DESCRIPTION
**Pull request by x09326**
PR for #9149 
@pvnr0082t 
Please help me review it, thanks.

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/young-steveo/bottlejs
 * Official website (optional, not the repository):
 * NPM package url (optional):https://www.npmjs.com/package/bottlejs
 * GitHub / Bitbucket popularity (required):
   - Count of stars:509
   - Count of watchers:19
   - Count of forks:27
 * NPM download stats (optional):
   - Downloads in the last day:210
   - Downloads in the last week:1214
   - Downloads in the last month:5878

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #9149, cc @young-steveo

### CODE REVIEW

File: ajax/libs/bottlejs/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "pimple"
+  ],
+  "author": "Stephen Young <young.steveo@gmail.com>",
+  "license": "MIT",
+  "npmName": "bottlejs",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/bottlejs/package.json]: @x09326 There is `bottle.d.ts` after v1.2.3. You should exclude it.

Comment (x09326) [inline ajax/libs/bottlejs/package.json]: Does not the ```.gitignore``` exclude it?

Comment (pvnr0082t) [inline ajax/libs/bottlejs/package.json]: We need to make sure we won't fetch this file type. Eventhough this type ignore by `.gitignore`.

Comment (x09326) [inline ajax/libs/bottlejs/package.json]: @pvnr0082t 
I have modified the file map.
Please help me review it.

Comment (pvnr0082t) [inline ajax/libs/bottlejs/package.json]: @x09326 `"**/!(*.ts)"` will be more flexible.

**Comment by PeterDaveHello** [comment]
Ping

**Comment by pvnr0082t** [comment]
@x09326 Sorry for late review. Please rebased it on the latest master branch.

**Comment by x09326** [comment]
@pvnr0082t 
I have rebased it.
Please help me review it, thanks.

**Comment by pvnr0082t** [comment]
LGTM

**Pull request merged**